bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/dialogs/CheckedTreeSelectionDialog.java

```diff
@@ -228,6 +228,16 @@ public void setSize(int width, int height) {
 		fHeight = height;
 	}
 
+	/**
+	 * Returns whether an empty tree is allowed
+	 *
+	 * @return true if empty tree is ok, false otherwise
+	 * @since 3.131
+	 */
+	protected boolean isEmptyTreeAllowed() {
```

Comment:
Do you want this flag to change for a given instance? If not, what about making it a constructor argument instead?

Reply:
I did not want the flag to change.  I'll add a new constructor as suggested.